com.konstantintutsch.Lock.yaml

```diff
@@ -0,0 +1,38 @@
+id: com.konstantintutsch.Lock
+runtime: org.gnome.Platform
+runtime-version: '47'
+sdk: org.gnome.Sdk
+command: com.konstantintutsch.Lock
+finish-args:
+  - --device=dri
+  - --share=ipc
+  - --socket=wayland
+  - --socket=fallback-x11
+  - --filesystem=xdg-run/gnupg:ro
+  - --filesystem=~/.gnupg:create
+  - --socket=gpg-agent
+  - --socket=session-bus
```

Comment:
try figuring out what exact bus names are required, `session-bus` is not allowed. You cn run the app with `flatpak run --nosocket=session-bus --log-session-bus`

Reply:
Is there anything I can look out for in the log? I'm a bit clueless at the moment.

When executing a step that requires some part of the session bus, my app fails but the log of the bus does not change. And when I then grant my app access to the real session bus again, that operation executes flawlessly. So it's definitely related to the bus, but I have no idea where I should examine the bus more closely to fix this 😅

I would be really thankful if you've got an idea how I could debug this

Reply:
can post the output of `--log-session-bus`?

Reply:
Sure, here you go:

```
** Message: 09:27:12.662: GnuPG Made Easy 1.23.2
C1: -> org.freedesktop.DBus call org.freedesktop.DBus.Hello at /org/freedesktop/DBus
C2: -> org.freedesktop.DBus fake wildcarded AddMatch for org.freedesktop.portal
C3: -> org.freedesktop.DBus fake wildcarded AddMatch for com.konstantintutsch.Lock
C4: -> org.freedesktop.DBus fake wildcarded AddMatch for org.mpris.MediaPlayer2.com.konstantintutsch.Lock
C5: -> org.freedesktop.DBus fake ListNames
B-1: <- org.freedesktop.DBus return from C1
B-1: <- org.freedesktop.DBus signal org.freedesktop.DBus.NameAcquired at /org/freedesktop/DBus
B-1: <- org.freedesktop.DBus return from C2
*SKIPPED*
B-1: <- org.freedesktop.DBus return from C3
*SKIPPED*
B-1: <- org.freedesktop.DBus return from C4
*SKIPPED*
B-1: <- org.freedesktop.DBus return from C5
C6: -> org.freedesktop.DBus fake GetNameOwner for org.freedesktop.DBus
C7: -> org.freedesktop.DBus fake GetNameOwner for org.freedesktop.portal.Desktop
C8: -> org.freedesktop.DBus fake GetNameOwner for org.freedesktop.portal.Documents
C9: -> org.freedesktop.DBus fake GetNameOwner for org.freedesktop.portal.IBus
*SKIPPED*
B-1: <- org.freedesktop.DBus return from C6
*SKIPPED*
B-1: <- org.freedesktop.DBus return from C7
*SKIPPED*
B-1: <- org.freedesktop.DBus return from C8
*SKIPPED*
B-1: <- org.freedesktop.DBus return from C9
*SKIPPED*
C10: -> org.freedesktop.DBus call org.freedesktop.DBus.AddMatch at /org/freedesktop/DBus
C11: -> org.freedesktop.DBus call org.freedesktop.DBus.StartServiceByName at /org/freedesktop/DBus
Filtering message due to arg0 org.gtk.vfs.Daemon, policy: 0 (required 2)
*HIDDEN* (ping)
B-1: <- org.freedesktop.DBus return from C10
B-1: <- org.freedesktop.DBus return from C11
*REWRITTEN*
C12: -> org.freedesktop.DBus call org.freedesktop.DBus.GetNameOwner at /org/freedesktop/DBus
Filtering message due to arg0 org.gtk.vfs.Daemon, policy: 0 (required 1)
B-1: <- org.freedesktop.DBus return from C12
*REWRITTEN*
C13: -> org.gtk.vfs.Daemon call org.gtk.vfs.MountTracker.ListMountableInfo at /org/gtk/vfs/mounttracker
*HIDDEN* (ping)
B-1: <- org.freedesktop.DBus return from C13
*REWRITTEN*
C14: -> org.freedesktop.DBus call org.freedesktop.DBus.RemoveMatch at /org/freedesktop/DBus
C1: -> org.freedesktop.DBus call org.freedesktop.DBus.Hello at /org/freedesktop/DBus
C2: -> org.freedesktop.DBus fake wildcarded AddMatch for org.freedesktop.portal
C3: -> org.freedesktop.DBus fake wildcarded AddMatch for com.konstantintutsch.Lock
C4: -> org.freedesktop.DBus fake wildcarded AddMatch for org.mpris.MediaPlayer2.com.konstantintutsch.Lock
C5: -> org.freedesktop.DBus fake ListNames
B-1: <- org.freedesktop.DBus return from C1
B-1: <- org.freedesktop.DBus signal org.freedesktop.DBus.NameAcquired at /org/freedesktop/DBus
B-1: <- org.freedesktop.DBus return from C2
*SKIPPED*
B-1: <- org.freedesktop.DBus return from C3
*SKIPPED*
B-1: <- org.freedesktop.DBus return from C4
*SKIPPED*
B-1: <- org.freedesktop.DBus return from C5
C6: -> org.freedesktop.DBus fake GetNameOwner for org.freedesktop.DBus
C7: -> org.freedesktop.DBus fake GetNameOwner for org.freedesktop.portal.Desktop
C8: -> org.freedesktop.DBus fake GetNameOwner for org.freedesktop.portal.Documents
C9: -> org.freedesktop.DBus fake GetNameOwner for org.freedesktop.portal.IBus
*SKIPPED*
B-1: <- org.freedesktop.DBus return from C6
*SKIPPED*
B-1: <- org.freedesktop.DBus return from C7
*SKIPPED*
B-1: <- org.freedesktop.DBus return from C8
*SKIPPED*
B-1: <- org.freedesktop.DBus return from C9
*SKIPPED*
C10: -> org.freedesktop.DBus call org.freedesktop.DBus.RequestName at /org/freedesktop/DBus
B-1: <- org.freedesktop.DBus signal org.freedesktop.DBus.NameOwnerChanged at /org/freedesktop/DBus
B-1: <- org.freedesktop.DBus signal org.freedesktop.DBus.NameAcquired at /org/freedesktop/DBus
B-1: <- org.freedesktop.DBus return from C10
C11: -> org.freedesktop.DBus call org.freedesktop.DBus.AddMatch at /org/freedesktop/DBus
C12: -> org.freedesktop.DBus call org.freedesktop.DBus.GetNameOwner at /org/freedesktop/DBus
C13: -> org.freedesktop.DBus call org.freedesktop.DBus.AddMatch at /org/freedesktop/DBus
C14: -> org.freedesktop.DBus call org.freedesktop.DBus.AddMatch at /org/freedesktop/DBus
C15: -> org.freedesktop.DBus call org.freedesktop.DBus.StartServiceByName at /org/freedesktop/DBus
B-1: <- org.freedesktop.DBus return from C11
B-1: <- org.freedesktop.DBus return from C12
B-1: <- org.freedesktop.DBus return from C13
B-1: <- org.freedesktop.DBus return from C14
B-1: <- org.freedesktop.DBus return from C15
C16: -> org.freedesktop.DBus call org.freedesktop.DBus.GetNameOwner at /org/freedesktop/DBus
B-1: <- org.freedesktop.DBus return from C16
C17: -> :1.89 call org.freedesktop.DBus.Properties.GetAll at /org/freedesktop/portal/desktop
B201: <- :1.89 return from C17
C18: -> :1.89 call org.freedesktop.portal.Settings.ReadAll at /org/freedesktop/portal/desktop
B205: <- :1.89 return from C18
C19: -> org.freedesktop.DBus call org.freedesktop.DBus.AddMatch at /org/freedesktop/DBus
C20: -> org.freedesktop.DBus call org.freedesktop.DBus.AddMatch at /org/freedesktop/DBus
C21: -> org.freedesktop.DBus call org.freedesktop.DBus.StartServiceByName at /org/freedesktop/DBus
B-1: <- org.freedesktop.DBus return from C19
B-1: <- org.freedesktop.DBus return from C20
B-1: <- org.freedesktop.DBus return from C21
C22: -> org.freedesktop.DBus call org.freedesktop.DBus.GetNameOwner at /org/freedesktop/DBus
B-1: <- org.freedesktop.DBus return from C22
C23: -> :1.89 call org.freedesktop.DBus.Properties.GetAll at /org/freedesktop/portal/desktop
B206: <- :1.89 return from C23
C24: -> org.freedesktop.DBus call org.freedesktop.DBus.StartServiceByName at /org/freedesktop/DBus
B-1: <- org.freedesktop.DBus return from C24
C25: -> org.freedesktop.DBus call org.freedesktop.DBus.GetNameOwner at /org/freedesktop/DBus
B-1: <- org.freedesktop.DBus return from C25
C26: -> :1.89 call org.freedesktop.DBus.Properties.GetAll at /org/freedesktop/portal/desktop
B207: <- :1.89 return from C26
C27: -> :1.89 call org.freedesktop.portal.Settings.Read at /org/freedesktop/portal/desktop
B209: <- :1.89 return from C27
C28: -> :1.89 call org.freedesktop.portal.Settings.Read at /org/freedesktop/portal/desktop
B211: <- :1.89 return from C28
C29: -> :1.89 call org.freedesktop.portal.Settings.Read at /org/freedesktop/portal/desktop
B214: <- :1.89 return error org.freedesktop.portal.Error.NotFound from C29
B512: <- :1.23 call org.gtk.Actions.DescribeAll at /com/konstantintutsch/Lock
B516: <- :1.23 call org.freedesktop.DBus.Properties.GetAll at /com/konstantintutsch/Lock
C30: -> :1.23 return from B512
C31: -> :1.23 return from B516
C32: -> org.freedesktop.portal.Flatpak call org.freedesktop.DBus.Properties.Get at /org/freedesktop/portal/Flatpak
B525: <- :1.23 call org.gtk.Actions.DescribeAll at /com/konstantintutsch/Lock/window/1
B-1: <- org.freedesktop.DBus signal org.freedesktop.DBus.NameOwnerChanged at /org/freedesktop/DBus
B11: <- :1.104 return from C32
C33: -> org.freedesktop.DBus call org.freedesktop.DBus.AddMatch at /org/freedesktop/DBus
C34: -> org.freedesktop.DBus call org.freedesktop.DBus.GetNameOwner at /org/freedesktop/DBus
B-1: <- org.freedesktop.DBus return from C33
B-1: <- org.freedesktop.DBus return from C34
C35: -> :1.23 return from B525
C1: -> org.freedesktop.DBus call org.freedesktop.DBus.Hello at /org/freedesktop/DBus
C2: -> org.freedesktop.DBus fake wildcarded AddMatch for org.freedesktop.portal
C3: -> org.freedesktop.DBus fake wildcarded AddMatch for com.konstantintutsch.Lock
C4: -> org.freedesktop.DBus fake wildcarded AddMatch for org.mpris.MediaPlayer2.com.konstantintutsch.Lock
C5: -> org.freedesktop.DBus fake ListNames
B-1: <- org.freedesktop.DBus return from C1
B-1: <- org.freedesktop.DBus signal org.freedesktop.DBus.NameAcquired at /org/freedesktop/DBus
B-1: <- org.freedesktop.DBus return from C2
*SKIPPED*
B-1: <- org.freedesktop.DBus return from C3
*SKIPPED*
B-1: <- org.freedesktop.DBus return from C4
*SKIPPED*
B-1: <- org.freedesktop.DBus return from C5
C6: -> org.freedesktop.DBus fake GetNameOwner for org.freedesktop.DBus
C7: -> org.freedesktop.DBus fake GetNameOwner for com.konstantintutsch.Lock
C8: -> org.freedesktop.DBus fake GetNameOwner for org.freedesktop.portal.Desktop
C9: -> org.freedesktop.DBus fake GetNameOwner for org.freedesktop.portal.Documents
C10: -> org.freedesktop.DBus fake GetNameOwner for org.freedesktop.portal.Flatpak
C11: -> org.freedesktop.DBus fake GetNameOwner for org.freedesktop.portal.IBus
*SKIPPED*
B-1: <- org.freedesktop.DBus return from C6
*SKIPPED*
B-1: <- org.freedesktop.DBus return from C7
*SKIPPED*
B-1: <- org.freedesktop.DBus return from C8
*SKIPPED*
B-1: <- org.freedesktop.DBus return from C9
*SKIPPED*
B-1: <- org.freedesktop.DBus return from C10
*SKIPPED*
B-1: <- org.freedesktop.DBus return from C11
*SKIPPED*
C12: -> org.freedesktop.DBus call org.freedesktop.DBus.AddMatch at /org/freedesktop/DBus
C13: -> org.freedesktop.DBus call org.freedesktop.DBus.GetNameOwner at /org/freedesktop/DBus
Filtering message due to arg0 org.gnome.keyring.SystemPrompter, policy: 0 (required 1)
C14: -> org.gnome.keyring.SystemPrompter call org.gnome.keyring.internal.Prompter.BeginPrompting at /org/gnome/keyring/Prompter
*HIDDEN* (ping)
B-1: <- org.freedesktop.DBus return from C12
B-1: <- org.freedesktop.DBus return from C13
*REWRITTEN*
B-1: <- org.freedesktop.DBus return from C14
*REWRITTEN*
C1: -> org.freedesktop.DBus call org.freedesktop.DBus.Hello at /org/freedesktop/DBus
C2: -> org.freedesktop.DBus fake wildcarded AddMatch for org.freedesktop.portal
C3: -> org.freedesktop.DBus fake wildcarded AddMatch for com.konstantintutsch.Lock
C4: -> org.freedesktop.DBus fake wildcarded AddMatch for org.mpris.MediaPlayer2.com.konstantintutsch.Lock
C5: -> org.freedesktop.DBus fake ListNames
B-1: <- org.freedesktop.DBus return from C1
B-1: <- org.freedesktop.DBus signal org.freedesktop.DBus.NameAcquired at /org/freedesktop/DBus
B-1: <- org.freedesktop.DBus return from C2
*SKIPPED*
B-1: <- org.freedesktop.DBus return from C3
*SKIPPED*
B-1: <- org.freedesktop.DBus return from C4
*SKIPPED*
B-1: <- org.freedesktop.DBus return from C5
C6: -> org.freedesktop.DBus fake GetNameOwner for org.freedesktop.DBus
C7: -> org.freedesktop.DBus fake GetNameOwner for com.konstantintutsch.Lock
C8: -> org.freedesktop.DBus fake GetNameOwner for org.freedesktop.portal.Desktop
C9: -> org.freedesktop.DBus fake GetNameOwner for org.freedesktop.portal.Documents
C10: -> org.freedesktop.DBus fake GetNameOwner for org.freedesktop.portal.Flatpak
C11: -> org.freedesktop.DBus fake GetNameOwner for org.freedesktop.portal.IBus
*SKIPPED*
B-1: <- org.freedesktop.DBus return from C6
*SKIPPED*
B-1: <- org.freedesktop.DBus return from C7
*SKIPPED*
B-1: <- org.freedesktop.DBus return from C8
*SKIPPED*
B-1: <- org.freedesktop.DBus return from C9
*SKIPPED*
B-1: <- org.freedesktop.DBus return from C10
*SKIPPED*
B-1: <- org.freedesktop.DBus return from C11
*SKIPPED*
C12: -> org.freedesktop.portal.Desktop call org.freedesktop.DBus.Properties.Get at /org/freedesktop/portal/desktop
B219: <- :1.89 return from C12
C1: -> org.freedesktop.DBus call org.freedesktop.DBus.Hello at /org/freedesktop/DBus
C2: -> org.freedesktop.DBus fake wildcarded AddMatch for org.freedesktop.portal
C3: -> org.freedesktop.DBus fake wildcarded AddMatch for com.konstantintutsch.Lock
C4: -> org.freedesktop.DBus fake wildcarded AddMatch for org.mpris.MediaPlayer2.com.konstantintutsch.Lock
C5: -> org.freedesktop.DBus fake ListNames
B-1: <- org.freedesktop.DBus return from C1
B-1: <- org.freedesktop.DBus signal org.freedesktop.DBus.NameAcquired at /org/freedesktop/DBus
B-1: <- org.freedesktop.DBus return from C2
*SKIPPED*
B-1: <- org.freedesktop.DBus return from C3
*SKIPPED*
B-1: <- org.freedesktop.DBus return from C4
*SKIPPED*
B-1: <- org.freedesktop.DBus return from C5
C6: -> org.freedesktop.DBus fake GetNameOwner for org.freedesktop.DBus
C7: -> org.freedesktop.DBus fake GetNameOwner for com.konstantintutsch.Lock
C8: -> org.freedesktop.DBus fake GetNameOwner for org.freedesktop.portal.Desktop
C9: -> org.freedesktop.DBus fake GetNameOwner for org.freedesktop.portal.Documents
C10: -> org.freedesktop.DBus fake GetNameOwner for org.freedesktop.portal.Flatpak
C11: -> org.freedesktop.DBus fake GetNameOwner for org.freedesktop.portal.IBus
*SKIPPED*
B-1: <- org.freedesktop.DBus return from C6
*SKIPPED*
B-1: <- org.freedesktop.DBus return from C7
*SKIPPED*
B-1: <- org.freedesktop.DBus return from C8
*SKIPPED*
B-1: <- org.freedesktop.DBus return from C9
*SKIPPED*
B-1: <- org.freedesktop.DBus return from C10
*SKIPPED*
B-1: <- org.freedesktop.DBus return from C11
*SKIPPED*
C12: -> org.freedesktop.DBus call org.freedesktop.DBus.AddMatch at /org/freedesktop/DBus
C13: -> org.freedesktop.DBus call org.freedesktop.DBus.StartServiceByName at /org/freedesktop/DBus
Filtering message due to arg0 org.gtk.vfs.Daemon, policy: 0 (required 2)
*HIDDEN* (ping)
B-1: <- org.freedesktop.DBus return from C12
B-1: <- org.freedesktop.DBus return from C13
*REWRITTEN*
C14: -> org.freedesktop.DBus call org.freedesktop.DBus.GetNameOwner at /org/freedesktop/DBus
Filtering message due to arg0 org.gtk.vfs.Daemon, policy: 0 (required 1)
B-1: <- org.freedesktop.DBus return from C14
*REWRITTEN*
C15: -> org.gtk.vfs.Daemon call org.gtk.vfs.MountTracker.ListMountableInfo at /org/gtk/vfs/mounttracker
*HIDDEN* (ping)
B-1: <- org.freedesktop.DBus return from C15
*REWRITTEN*
C16: -> org.freedesktop.DBus call org.freedesktop.DBus.RemoveMatch at /org/freedesktop/DBus
C1: -> org.freedesktop.DBus call org.freedesktop.DBus.Hello at /org/freedesktop/DBus
C2: -> org.freedesktop.DBus fake wildcarded AddMatch for org.freedesktop.portal
C3: -> org.freedesktop.DBus fake wildcarded AddMatch for com.konstantintutsch.Lock
C4: -> org.freedesktop.DBus fake wildcarded AddMatch for org.mpris.MediaPlayer2.com.konstantintutsch.Lock
C5: -> org.freedesktop.DBus fake ListNames
B-1: <- org.freedesktop.DBus return from C1
B-1: <- org.freedesktop.DBus signal org.freedesktop.DBus.NameAcquired at /org/freedesktop/DBus
B-1: <- org.freedesktop.DBus return from C2
*SKIPPED*
B-1: <- org.freedesktop.DBus return from C3
*SKIPPED*
B-1: <- org.freedesktop.DBus return from C4
*SKIPPED*
B-1: <- org.freedesktop.DBus return from C5
C6: -> org.freedesktop.DBus fake GetNameOwner for org.freedesktop.DBus
C7: -> org.freedesktop.DBus fake GetNameOwner for com.konstantintutsch.Lock
C8: -> org.freedesktop.DBus fake GetNameOwner for org.freedesktop.portal.Desktop
C9: -> org.freedesktop.DBus fake GetNameOwner for org.freedesktop.portal.Documents
C10: -> org.freedesktop.DBus fake GetNameOwner for org.freedesktop.portal.Flatpak
C11: -> org.freedesktop.DBus fake GetNameOwner for org.freedesktop.portal.IBus
*SKIPPED*
B-1: <- org.freedesktop.DBus return from C6
*SKIPPED*
B-1: <- org.freedesktop.DBus return from C7
*SKIPPED*
B-1: <- org.freedesktop.DBus return from C8
*SKIPPED*
B-1: <- org.freedesktop.DBus return from C9
*SKIPPED*
B-1: <- org.freedesktop.DBus return from C10
*SKIPPED*
B-1: <- org.freedesktop.DBus return from C11
*SKIPPED*
C12: -> org.freedesktop.DBus call org.freedesktop.DBus.AddMatch at /org/freedesktop/DBus
C13: -> org.freedesktop.DBus call org.freedesktop.DBus.GetNameOwner at /org/freedesktop/DBus
C14: -> org.freedesktop.portal.Desktop call org.freedesktop.portal.Secret.RetrieveSecret at /org/freedesktop/portal/desktop
B-1: <- org.freedesktop.DBus return from C12
B-1: <- org.freedesktop.DBus return from C13
B224: <- :1.89 return from C14
B226: <- :1.89 signal org.freedesktop.portal.Request.Response at /org/freedesktop/portal/desktop/request/1_108/libsecret2043691700
C15: -> org.freedesktop.DBus call org.freedesktop.DBus.RemoveMatch at /org/freedesktop/DBus
B-1: <- org.freedesktop.DBus return from C15

** (com.konstantintutsch.Lock:2): WARNING **: 09:27:24.464: Failed to decrypt GPGME data from memory: No such file or directory
```

---

The `No such file or directory` error is from `gpgme_op_decrypt()`. It only occurs when the secret key is not decrypted and a pinentry is required.

---

Adding

- `--talk-name=org.freedesktop.portal.Secret.RetrieveSecret`
- `--talk-name=org.freedesktop.portal.Request.Response`

did not fix the bug.

Reply:
one of them is `--talk-name=org.gnome.keyring.SystemPrompter`, those portal ones are automatically allowed.

Reply:
Mmmh, okay. I've had a look at the manifests of other applications interacting with GPG and found a fix in [org.gnupg.GPA](https://github.com/flathub/org.gnupg.GPA/blob/4cf379db1e677670c2b544a393cd7ee4b9065483/org.gnupg.GPA.yaml#L35).

Their manifest includes Pinentry in the Flatpak itself. That's what I've done too now.

Works flawlessly 👌